selfdrive/common/clutil.cc

```diff
@@ -0,0 +1,185 @@
+#include "clutil.h"
+#include <assert.h>
+#include <inttypes.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <memory>
+#include <iostream>
+#include <vector>
+#include "util.h"
+#include "utilpp.h"
+
+namespace {  // helper functions
+
+template <typename Func, typename Id, typename Name>
+std::string get_info(Func get_info_func, Id id, Name param_name) {
+  size_t size = 0;
+  CL_CHECK(get_info_func(id, param_name, 0, NULL, &size));
+  std::string info(size, '\0');
+  CL_CHECK(get_info_func(id, param_name, size, &info[0], NULL));
+  return info;
+}
+inline std::string get_platform_info(cl_platform_id id, cl_platform_info name) { return get_info(&clGetPlatformInfo, id, name); }
+inline std::string get_device_info(cl_device_id id, cl_device_info name) { return get_info(&clGetDeviceInfo, id, name); }
+
+void cl_print_info(cl_platform_id platform, cl_device_id device) {
+  size_t work_group_size = 0;
+  cl_device_type device_type = 0;
+  clGetDeviceInfo(device, CL_DEVICE_MAX_WORK_GROUP_SIZE, sizeof(work_group_size), &work_group_size, NULL);
+  clGetDeviceInfo(device, CL_DEVICE_TYPE, sizeof(device_type), &device_type, NULL);
+  const char *type_str = "Other...";
+  switch (device_type) {
+    case CL_DEVICE_TYPE_CPU: type_str ="CL_DEVICE_TYPE_CPU"; break;
+    case CL_DEVICE_TYPE_GPU: type_str = "CL_DEVICE_TYPE_GPU"; break;
+    case CL_DEVICE_TYPE_ACCELERATOR: type_str = "CL_DEVICE_TYPE_ACCELERATOR"; break;
+  }
+
+  std::cout << "vendor: " << get_platform_info(platform, CL_PLATFORM_VENDOR) << std::endl
+            << "platform version: " << get_platform_info(platform, CL_PLATFORM_VERSION) << std::endl
+            << "profile: " << get_platform_info(platform, CL_PLATFORM_PROFILE) << std::endl
+            << "extensions: " << get_platform_info(platform, CL_PLATFORM_EXTENSIONS) << std::endl
+            << "name :" << get_device_info(device, CL_DEVICE_NAME) << std::endl
+            << "device version :" << get_device_info(device, CL_DEVICE_VERSION) << std::endl
+            << "max work group size :" << work_group_size << std::endl
+            << "type = " << std::hex << std::showbase << device_type << " = " << type_str << std::endl;
```

Comment:
You're not restoring the output properties of `std::cout`. Won't everything after this be in hex? Printing in hex is not that important, so maybe just leave it as is?